pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,157 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
+	// +optional
+	MinDomains *int32 `json:"minDomains,omitempty" protobuf:"varint,5,opt,name=minDomains"`
+	// NodeAffinityPolicy indicates how we will treat Pod's nodeAffinity/nodeSelector
+	// when calculating pod topology spread skew. Options are:
+	// - Honor: only nodes matching nodeAffinity/nodeSelector are included in the calculations.
+	// - Ignore: nodeAffinity/nodeSelector are ignored. All nodes are included in the calculations.
+	//
+	// If this value is nil, the behavior is equivalent to the Honor policy.
+	// This is a beta-level feature default enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
+	// +optional
+	NodeAffinityPolicy *v1.NodeInclusionPolicy `json:"nodeAffinityPolicy,omitempty" protobuf:"bytes,6,opt,name=nodeAffinityPolicy"`
+	// NodeTaintsPolicy indicates how we will treat node taints when calculating
+	// pod topology spread skew. Options are:
+	// - Honor: nodes without taints, along with tainted nodes for which the incoming pod
+	// has a toleration, are included.
+	// - Ignore: node taints are ignored. All nodes are included.
+	//
+	// If this value is nil, the behavior is equivalent to the Ignore policy.
+	// This is a beta-level feature default enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
+	// +optional
+	NodeTaintsPolicy *v1.NodeInclusionPolicy `json:"nodeTaintsPolicy,omitempty" protobuf:"bytes,7,opt,name=nodeTaintsPolicy"`
+	// MatchLabelKeys is a set of pod label keys to select the pods over which
+	// spreading will be calculated. The keys are used to lookup values from the
+	// incoming pod labels, those key-value labels are ANDed with labelSelector
+	// to select the group of existing pods over which spreading will be calculated
+	// for the incoming pod. The same key is forbidden to exist in both MatchLabelKeys and LabelSelector.
+	// MatchLabelKeys cannot be set when LabelSelector isn't set.
+	// Keys that don't exist in the incoming pod labels will
+	// be ignored. A null or empty list means only match against labelSelector.
+	//
+	// This is a beta field and requires the MatchLabelKeysInPodTopologySpread feature gate to be enabled (enabled by default).
+	// +listType=atomic
+	// +optional
+	MatchLabelKeys []string `json:"matchLabelKeys,omitempty" protobuf:"bytes,8,opt,name=matchLabelKeys"`
+
+	//+optional
+	// +listType=atomic
+	// Defines what Prometheus Operator managed labels should be added to topologySpreadConstraints[].labelSelector.
+	AdditionalLabelSelectors []string `json:"additionalLabelSelectors,omitempty"`
+}
+
+func (tp TopologySpreadConstraint) BuildK8sTopologySpreadConstraint(podLabels map[string]string) v1.TopologySpreadConstraint {
+
+	if len(tp.AdditionalLabelSelectors) > 0 {
+		if tp.LabelSelector == nil {
+			tp.LabelSelector = &metav1.LabelSelector{
+				MatchLabels: make(map[string]string),
+			}
+		}
+
+		for _, label := range tp.AdditionalLabelSelectors {
+			tp.LabelSelector.MatchLabels[label] = podLabels[label]
```

Comment:
I'm not sure what should be the behavior if users provide a label that dont exists on the podLabels.
Should we fail or just log it?

Reply:
Another question, is it possible to override pre-existing labels here? If it is, is that a problem?

Reply:
Not really!
Because there are no default labels, it's filled by the user

Reply:
aah ok